includes/abstracts/SimpleUser/SimpleUser.php

```diff
@@ -73,13 +73,21 @@ public function __construct($username, $password, $require_password=true)
 		{
 			$this->install();
 		}
-		if($this->_login($username, $password, $require_password))
-		{
-			$this->is_logged_in=true;
-			$this->uid=$this->userInfo->getId();
-			$this->is_admin=$this->userInfo->getIs_admin();
-			$this->username=$username;
-			$this->password=$password;
+		if($require_password) {
```

Comment:
We could probably simplify this a bit - why doesn't _login() do the check for require_password?